src/test/java/com/ncc/aif/ScalingTest.java

```diff
@@ -259,62 +326,114 @@ private void addEventOrRelation() {
+        if (preserveValidity(NIST_RESTRICTION.RestrictJustification_Edges)) {
+            addEdgeJustificationAndPrivateData(argument);
+        } else {
+            addJustificationAndPrivateData(argument);
+        }
 
-        Resource argumentTwo = markAsArgument(model, eventResource,
-                SeedlingOntology.Physical_Resident_Resident,
+        Resource argumentTwo = markAsArgument(model, relationResource,
+                LDCOntology.Physical_Resident_Resident_Resident,
                 getRandomEntity(), system, 0.785, getAssertionUri());
-        addJustificationAndPrivateData(argumentTwo);
+        addEdgeJustificationAndPrivateData(argumentTwo);
     }
 
     private void addJustificationAndPrivateData(Resource resource) {
         String docId = getRandomDocId();
 
         // Justify the type assertion
-        markTextJustification(model, ImmutableSet.of(resource), docId, 1029, 1033, system, 0.973);
+        if (preserveValidity(NIST_RESTRICTION.JustifyTypeAssertions)) {
+            markTextJustification(model, resource, docId, 1029, 1033, system,
+                    preserveValidity(NIST_RESTRICTION.ConfidenceValueRange) ? 0.973 : 1.973);
+        }
 
         // Add some private data
         markPrivateData(model, resource, "{ 'provenance' : '" + docId + "' }", system);
     }
 
-    private final ValidateAIF seedlingValidator = ValidateAIF.createForDomainOntologySource(
-            Resources.asCharSource(Resources.getResource("com/ncc/aif/ontologies/SeedlingOntology"), StandardCharsets.UTF_8));
+    private void addEdgeJustificationAndPrivateData(Resource resource) {
+        String docId = getRandomDocId();
+
+        // Justify the edge
+        final Resource justification = preserveValidity(NIST_RESTRICTION.PreventShotVideo) ?
+                makeTextJustification(model, docId, 1029, 1033, system, 0.973) :
+                markShotVideoJustification(model, getRandomEntity(), "source1", "shotId", system, 1.0);
+        Resource compound;
+        if (checkChaos(NIST_RESTRICTION.EdgeJustificationLimit)) {
+            final Resource justification2 = makeTextJustification(model, docId, 1055, 1071, system, 0.677);
+            final Resource justification3 = makeTextJustification(model, docId, 1102, 1159, system, 0.881);
+            compound = markCompoundJustification(model,
+                    ImmutableSet.of(resource), ImmutableSet.of(justification, justification2, justification3), system, 1.0);
+        } else {
+            compound = markCompoundJustification(model,
+                    ImmutableSet.of(resource), ImmutableSet.of(justification), system,
+                    preserveValidity(NIST_RESTRICTION.ConfidenceValueRange) ? 1.0 : -1.0);
+        }
+        markJustification(resource, compound);
+
+        // Add some private data
+        markPrivateData(model, resource, "{ 'provenance' : '" + docId + "' }", system);
+    }
+
+    // Returns true when invalid content should be generated
```

Comment:
These two methods are confusing.
Why is the comment on checkChaos identical to the comment on preserveValidity when they are doing opposite things (I think).
To me, the method name "checkChaos" doesn't imply the opposite of preserveValidity.

Reply:
The comment was definitely a copy/paste error-- fixed.
I changed checkChaos to introduceChaos.  I'm open to suggestions.  `introduceInvalidity()`?  `introduceInvalidContent()`?

Reply:
introduceChaos is much better